components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1472,6 +1472,9 @@ private UsersGetResponse filterUsersBySingleAttribute(ExpressionNode node, Map<S
         domainName = resolveDomainName(domainName, node);
         int totalResults = 0;
         try {
+            if (limit == 0) {
+                limit = getMaxLimit(domainName);
+            }
```

Comment:
Fix the 4th problem mentioned here https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/609#issuecomment-2644715275

When the count is not provided, set the max user limit set for the given user store domain.